eventuals/task.h

```diff
@@ -19,13 +20,42 @@ namespace eventuals {
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename E_, typename From_, typename To_>
+template <typename>
+struct VariantErrorsHelper;
+
+template <typename... Errors>
+struct VariantErrorsHelper<std::variant<std::monostate, Errors...>> {
+  using type = std::conditional_t<
+      sizeof...(Errors) == 0,
+      void,
+      std::variant<Errors...>>;
+};
+
+////////////////////////////////////////////////////////////////////////
+
+template <
+    typename E_,
+    typename From_,
+    typename To_,
+    typename Catches_,
+    typename Raises_>
 struct HeapTask final {
+  using StartCallback_ = Callback<function_type_t<void, To_>>;
```

Comment:
Just like in `generator.h`, let's move these to the top-level so that we an reuse them multiple places!